local enums = {}
enums.__index = enums

export type harmProbability = {
    HARM_PROBABILITY_UNSPECIFIED: "HARM_PROBABILITY_UNSPECIFIED",
    NEGLIGIBLE: "NEGLIGIBLE",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    HIGH: "HIGH",
}

export type harmBlockThreshold = {
    HARM_BLOCK_THRESHOLD_UNSPECIFIED: "HARM_BLOCK_THRESHOLD_UNSPECIFIED",
    BLOCK_LOW_AND_ABOVE: "BLOCK_LOW_AND_ABOVE",
    BLOCK_MEDIUM_AND_ABOVE: "BLOCK_MEDIUM_AND_ABOVE",
    BLOCK_ONLY_HIGH: "BLOCK_ONLY_HIGH",
    BLOCK_NONE: "BLOCK_NONE",
    OFF: "OFF",
}

export type harmCategory = {
    HARM_CATEGORY_UNSPECIFIED: "HARM_CATEGORY_UNSPECIFIED",
    HARM_CATEGORY_DEROGATORY: "HARM_CATEGORY_DEROGATORY",
    HARM_CATEGORY_TOXICITY: "HARM_CATEGORY_TOXICITY",
    HARM_CATEGORY_VIOLENCE: "HARM_CATEGORY_VIOLENCE",
    HARM_CATEGORY_SEXUAL: "HARM_CATEGORY_SEXUAL",
    HARM_CATEGORY_MEDICAL: "HARM_CATEGORY_MEDICAL",
    HARM_CATEGORY_DANGEROUS: "HARM_CATEGORY_DANGEROUS",
    HARM_CATEGORY_HARASSMENT: "HARM_CATEGORY_HARASSMENT",
    HARM_CATEGORY_HATE_SPEECH: "HARM_CATEGORY_HATE_SPEECH",
    HARM_CATEGORY_SEXUALLY_EXPLICIT: "HARM_CATEGORY_SEXUALLY_EXPLICIT",
    HARM_CATEGORY_DANGEROUS_CONTENT: "HARM_CATEGORY_DANGEROUS_CONTENT",
    HARM_CATEGORY_CIVIC_INTEGRITY: "HARM_CATEGORY_CIVIC_INTEGRITY",
}

export type blockReason = {
    SAFETY: "SAFETY",
    OTHER: "OTHER",
    BLOCKLIST: "BLOCKLIST",
    PROHIBITED_CONTENT: "PROHIBITED_CONTENT",
}

export type finishReason = {
    STOP: "STOP",
    MAX_TOKENS: "MAX_TOKENS",
    SAFETY: "SAFETY",
    RECITATION: "RECITATION",
    LANGUAGE: "LANGUAGE",
    OTHER: "OTHER",
    BLOCKLIST: "BLOCKLIST",
    PROHIBITED_CONTENT: "PROHIBITED_CONTENT",
    SPII: "SPII",
    MALFORMED_FUNCTION_CALL: "MALFORMED_FUNCTION_CALL",
}

return enums