local enums = {}
enums.__index = enums

export type harmProbability = {
    string
}

export type harmBlockThreshold = {
    string
}

export type harmCategory = {
    string
}

export type blockReason = {
    string
}

export type finishReason = {
    string
}

enums.harmProbability = {
    HARM_PROBABILITY_UNSPECIFIED = "HARM_PROBABILITY_UNSPECIFIED",
    NEGLIGIBLE = "NEGLIGIBLE",
    LOW = "LOW",
    MEDIUM = "MEDIUM",
    HIGH = "HIGH",
}

enums.harmBlockThreshold = {
    BLOCK_LOW_AND_ABOVE = "BLOCK_LOW_AND_ABOVE",
    BLOCK_MEDIUM_AND_ABOVE = "BLOCK_MEDIUM_AND_ABOVE",
    BLOCK_ONLY_HIGH = "BLOCK_ONLY_HIGH",
    BLOCK_NONE = "BLOCK_NONE",
    OFF = "OFF",
}

enums.harmCategory = {
    UNSPECIFIED = "HARM_CATEGORY_UNSPECIFIED",
    DEROGATORY = "HARM_CATEGORY_DEROGATORY",
    TOXICITY = "HARM_CATEGORY_TOXICITY",
    VIOLENCE = "HARM_CATEGORY_VIOLENCE",
    SEXUAL = "HARM_CATEGORY_SEXUAL",
    MEDICAL = "HARM_CATEGORY_MEDICAL",
    DANGEROUS = "HARM_CATEGORY_DANGEROUS",
    HARASSMENT = "HARM_CATEGORY_HARASSMENT",
    HATE_SPEECH = "HARM_CATEGORY_HATE_SPEECH",
    SEXUALLY_EXPLICIT = "HARM_CATEGORY_SEXUALLY_EXPLICIT",
    DANGEROUS_CONTENT = "HARM_CATEGORY_DANGEROUS_CONTENT",
    CIVIC_INTEGRITY = "HARM_CATEGORY_CIVIC_INTEGRITY",
}

enums.blockReason = {
    SAFETY = "SAFETY",
    OTHER = "OTHER",
    BLOCKLIST = "BLOCKLIST",
    PROHIBITED_CONTENT = "PROHIBITED_CONTENT",
}

enums.finishReason = {
    STOP = "STOP",
    MAX_TOKENS = "MAX_TOKENS",
    SAFETY = "SAFETY",
    RECITATION = "RECITATION",
    LANGUAGE = "LANGUAGE",
    OTHER = "OTHER",
    BLOCKLIST = "BLOCKLIST",
    PROHIBITED_CONTENT = "PROHIBITED_CONTENT",
    SPII = "SPII",
    MALFORMED_FUNCTION_CALL = "MALFORMED_FUNCTION_CALL",
}

return enums